libs/dfi/src/dyn_type.c

```diff
@@ -185,265 +187,200 @@ static int dynType_parseAny(FILE *stream, dyn_type *type) {
     return status;
 }
 
-static int dynType_parseMetaInfo(FILE *stream, dyn_type *type) {
+static int dynType_parseMetaInfo(FILE* stream, dyn_type* type) {
     int status = OK;
-    char *name = NULL;
-    char *value = NULL;
-
-    struct meta_entry *entry = calloc(1, sizeof(*entry));
-    if (entry == NULL) {
-        status = ERROR;
-    }
+    celix_autofree char* name = NULL;
+    celix_autofree char* value = NULL;
 
-    if (status == OK) {
-        status = dynCommon_parseName(stream, &name);
+    if (dynCommon_parseName(stream, &name) != OK) {
+        status = PARSE_ERROR;
+        goto bail_out;
```

Comment:
goto is needed to reuse the celix_err_push(msg) + return status?

Reply:
Yes. Now I realize that there is a better way to do this.